### ISSUE TITLE
iOS 17 Beta Image display abnormally when the image has alpha channel.

### ISSUE THREAD

**Issue by 123sunxiaolin** [SDWebImage/SDWebImage#3566]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by 123sunxiaolin** [comment]
The code is :

+ (UIImage *)decodedImageWithImage:(UIImage *)image samplingSize:(CGSize)samplingSize {
...
 BOOL hasAlpha = [self CGImageContainsAlpha:imageRef];
//        if(hasAlpha) return image;
        
        // iOS prefer BGRA8888 (premultiplied) or BGRX8888 bitmapInfo for screen rendering, which is same as `UIGraphicsBeginImageContext()` or `- [CALayer drawInContext:]`
        // Though you can use any supported bitmapInfo (see: https://developer.apple.com/library/content/documentation/GraphicsImaging/Conceptual/drawingwithquartz2d/dq_context/dq_context.html#//apple_ref/doc/uid/TP30001066-CH203-BCIBHHBB ) and let Core Graphics reorder it when you call `CGContextDrawImage`
        // But since our build-in coders use this bitmapInfo, this can have a little performance benefit
        CGBitmapInfo bitmapInfo = kCGBitmapByteOrder32Host;  //BGRA8888 (premultiplied) or BGRX8888
        bitmapInfo |= (hasAlpha ? kCGImageAlphaFirst : kCGImageAlphaNoneSkipFirst);  // BGRA8888
...
}

**Comment by weihaihan066078** [comment]
这个应该是CoreGraphics的bug

**Comment by dreampiggy** [comment]
> I don't think this code exists on 5.17.0, try that firstly ?

Try again. I don't guranteed old version can work (which using the BGRA8888)
